### PR TITLE
Fix KubeStateMetricsDown alert

### PR DESCRIPTION
**Pull request by mcbenjemaa**
Towards https://github.com/giantswarm/giantswarm/issues/22262

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml [hunk 1]
```diff
@@ -48,34 +48,16 @@ spec:
       annotations:
         description: '{{`KubeStateMetrics ({{ $labels.instance }}) is down.`}}'
         opsrecipe: kube-state-metrics-down/
-      expr: label_replace(up{app="kube-state-metrics"}, "ip", "$1.$2.$3.$4", "node", "ip-(\\d+)-(\\d+)-(\\d+)-(\\d+).*") == 0
+      expr: label_replace(up{app="kube-state-metrics"}, "ip", "$1.$2.$3.$4", "node", "ip-(\\d+)-(\\d+)-(\\d+)-(\\d+).*") == 0  or absent(up{app="kube-state-metrics"} == 1)
```
Comment (TheoBrigitte) [inline helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: What's this magic with the node ip ? Do we need to propagate this ?

Comment (mcbenjemaa) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: I'm not sure.. .

Comment (mcbenjemaa) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: all up alerts are the same

Comment (mcbenjemaa) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: Do we merge?

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: I think this label might be used to match against other metrics, so I believe we should keep it.

Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: I did not check, but I'd bet on matching inhibitions.

Comment (mcbenjemaa) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: @QuentinBisson  has more knowledge about this

Comment (hervenicol) [reply helm/prometheus-rules/templates/alerting-rules/up.all.rules.yml:51]: I confirm it is inhibitions:

```
- source_matchers:
  - kubelet_down="true"
  target_matchers:
  - cancel_if_kubelet_down="true"
  equal:
  - cluster_id
  - ip
```
and:
```
- source_matchers:
  - kubelet_not_ready="true"
  target_matchers:
  - cancel_if_kubelet_not_ready="true"
  equal:
  - cluster_id
  - ip
```

**Comment by hervenicol** [comment]
I think we perverted the way inhibitions should work.

We have a 2 checks that looks for `kube-state-metrics` status:
* One called `InhibitionKubeStateMetricsDown`.
* One called `KubeStateMetricsDown`.

:shrug: That's the way all of our inhibitions work. I guess it's supposed to make it easier to read.
But I don't see why "inhibition" checks should not page.

**Review by hervenicol — APPROVED** [review]
LGTM.
I have concerns, but on a broader level.
No reasons to mix them with this PR :+1:

**Pull request merged**